develop/CICD/Tutorials/CICD_For_Other_Items/CICD_Tutorial_For_Other_Items_Multi-Artifact_DataMiner_Package_VisualStudio_And_GitHub.md

```diff
@@ -0,0 +1,214 @@
+
+   1. Select the *DataMiner Automation Script Library Project* template, and click *Next*.
+
+   1. Enter the project name `TutorialLibrary`.
+
+   1. Fill in your name as the author.
+
+   1. Click *Create*.
+
+1. Add a DataMiner Ad Hoc Data Source project:
+
+   1. In the Solution Explorer, at the very top, right-click the solution *TutorialPackage* and select *Add > new project*.
+
+   1. Select the *DataMiner Ad Hoc Data Source Project* template, and click *Next*.
+
+   1. Enter the project name `TutorialDataSource`.
+
+   1. Fill in your name as the author.
+
+   1. Click *Create*.
+
+## Step 2: Add installation-specific code
+
+In this step, you will focus on the extra code that should be executed when the content is installed.
+
+1. Add the **setup content**:
+
+   1. In the Solution Explorer, navigate to the project *TutorialPackage* and right-click the *SetupContent* folder.
+
+   1. Select *Add > New Item*.
+
+   1. Select to a new text file (.txt), and give it the name `MyConfig.txt`.
+
+   1. Write `Hello World!` as the content and save the file.
+
+1. Write the **installation code**:
+
+   1. In the Solution Explorer, navigate to the project *TutorialPackage* and double-click to open the *TutorialPackage.cs* file.
+
+   1. Change the content of the *Install* method to the following:
+
+      ```csharp
+         try
+         {
+            engine.Timeout = new TimeSpan(0, 10, 0);
+            engine.GenerateInformation("Starting installation");
+            var installer = new AppInstaller(Engine.SLNetRaw, context);
+            installer.InstallDefaultContent();
+   
+            string setupContentPath = installer.GetSetupContentDirectory();
+   
+            // Custom installation logic can be added here for each individual install package.
+            string pathToConfig = System.IO.Path.Combine(setupContentPath, "MyConfig.txt");
+            var config = System.IO.File.ReadAllText(pathToConfig);
+            engine.GenerateInformation($"Tutorial installation ran with config: {config}");
+         }
+         catch (Exception e)
+         {
+            engine.ExitFail($"Exception encountered during installation: {e}");
+         }
+      ```
+
+   1. Save the changes
+
+## Step 3: Create a GitHub repository
+
+1. In the menu bar of Visual Studio, select *GIT* > *Create GIT Repository*.
+
+1. Create a new GitHub repository:
+
+   1. Select a name, specify your GitHub account, and mark yourself as the owner.
+
+   1. When you have entered all the information, click *Create and Push*.
+
+1. Go to the newly created GitHub repository on <https://github.com/>.
+
+   > [!TIP]
+   > In Visual Studio 2022, you can open the *GIT* menu and select *GitHub/View Pull Requests* to quickly access the correct repository.
+
+## Step 4: Check the GitHub Actions
+
+1. Create a GitHub repository by going to **Git > Create Git Repository**, selecting GitHub, and filling in the wizard before clicking **Create and Push**.
```

Comment:
Is it correct that you need to create a GitHub repo again as part of this step (even though this was already done in step 3), or should this be removed here?

Reply:
Ah no, that's a copy-paste error. Please remove